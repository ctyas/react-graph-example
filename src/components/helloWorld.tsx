import { Query, QueryResult } from "react-apollo";
import { helloWorldQuery } from "./helloWorld.query";
import * as React from "react";

interface Result {
    hello: string
}

export const HelloWorldRenderer = ({ loading, error, data }: Pick<QueryResult<Result>, "loading" | "error" | "data">) => {
    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error :(</p>;
    if (data) return <div>
        <p>{data.hello}</p>
    </div>

    return null;
}

export const HelloWorld = () => (
    <Query<Result> query={helloWorldQuery}>
        {HelloWorldRenderer}
    </Query>
)