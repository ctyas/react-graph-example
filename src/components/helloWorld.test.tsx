import React from 'react';
import ReactDOM from 'react-dom';
import { HelloWorld, HelloWorldRenderer } from './HelloWorld';
import { MockedProvider, MockedResponse } from 'react-apollo/test-utils';
import { helloWorldQuery } from './helloWorld.query';
import { shallow } from "enzyme";

const mocks: MockedResponse[] = [
    {
        request: {
            query: helloWorldQuery,
            variables: {
                name: "World",
            }
        },
        result: {
            data: {
                hello: "Hello World!"
            }
        }
    }
]

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MockedProvider mocks={mocks} addTypename={false}><HelloWorld /></MockedProvider>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("renders loading", () => {
    const wrapper = shallow(<HelloWorldRenderer loading={true} data={undefined}/>);
    expect(wrapper).toMatchSnapshot();
})

it("renders an error", () => {
    const wrapper = shallow(<HelloWorldRenderer loading={false} error={{message: "Error"} as any} data={undefined}/>);
    expect(wrapper).toMatchSnapshot();
})

it("renders the result", () => {
    const wrapper = shallow(<HelloWorldRenderer loading={false} data={{ hello: "Hello World!"}}/>);
    expect(wrapper).toMatchSnapshot();
})

it("Renders null for unexpected args", () => {
    const actual = HelloWorldRenderer({ loading: false, data: undefined });
    expect(actual).toBeNull();
})

