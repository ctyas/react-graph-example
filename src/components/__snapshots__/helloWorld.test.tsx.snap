// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an error 1`] = `
<p>
  Error :(
</p>
`;

exports[`renders loading 1`] = `
<p>
  Loading...
</p>
`;

exports[`renders the result 1`] = `
<div>
  <p>
    Hello World!
  </p>
</div>
`;
